src/base/utils/string.cpp

```diff
@@ -198,3 +198,12 @@ TriStateBool Utils::String::parseTriStateBool(const QString &string)
         return TriStateBool::False;
     return TriStateBool::Undefined;
 }
+
+QString Utils::String::join(const QVector<QStringRef> &strings, const QString &separator)
+{
+    QString ret;
+    for (const QStringRef &str : strings)
+        ret += (str + separator);
+    ret.chop(separator.size());
```

Comment:
Too clumsy implementation. That's the first thing that came into your head, and you don't want to redo it?..
```c++
if (strings.count() == 0) return "";

QString ret = strings[0];
for (int i = 1; i < strings.count(); ++i)
    ret += strings[i];

return ret;
```

Reply:
>That's the first thing that came into your head, and you don't want to redo it?..

Where tf did you read that?

Reply:
Changed.

Reply:
>Where tf did you read that?

I'm sorry, I didn't mean to offend you, but didn't you stumble yourself as over that illogical solution?

Reply:
I just want to be clear I've never said that. 

>didn't you stumble yourself as over that illogical solution?

I felt weird for a second, but didn't pursuit it for unknown reason...